posts/inside-rust/2025-03-07-inferred-const-generic-arguments.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: "Inferred const generic arguments: Call for Testing!"
+author: BoxyUwU
+team: The Const Generics Project Group <https://rust-lang.github.io/project-const-generics/>
+---
+
+We are excited to announce that `feature(generic_arg_infer)` is nearing the point of stabilization. In this post we'd like to talk a bit about what this feature does, and what comes next for it.
+
+## What is `feature(generic_arg_infer)`
+
+When `feature(min_const_generics)` was [stabilized in early 2021](https://github.com/rust-lang/rust/pull/79135) it did not include the ability to use `_` as an explicit const argument:
+```rust
+fn foo() {
+  // This errors due to `_` as an array length being unsupported
+  let a: [u8; _] = [Default::default()];
+  // This is legal as `_` is permitted as a type argument
+  let b: [_; 1] = a;
+}
+```
+
+This is entirely a syntactic limitation; writing `_` is disallowed but it is still possible to elide generic argument listings that may involve const arguments:
+```rust
+fn foo<const N: usize>(_: [u8; N]) {}
+
+fn bar() {
+  // This errors due to `_` as a const argument being unsupported
+  foo::<_>([1]);
+  // This is legal as even though the const argument is *inferred*
+  // there is no explicit `_` written.
+  foo([1]);
+}
+```
+
+The compiler has always been able to infer values for const generic parameters, only the ability to explicitly ask for a const argument to be inferred is unstable.
+
+`feature(generic_arg_infer)`  also allows you to infer the length of a repeat expression. This was previously not possible without moving the expression into a separate function generic over the array length.
+
+```rust
+fn foo() {
+    // This errors due to `_` as a repeat count being unsupported
+    let a: [_; 1] = [String::new(); _];
+}
+```
+
+This feature allows all of the previous examples to compile, which should hopefully feel like something that should "obviously" be supported by Rust.
```

Comment:
````suggestion
It is currently also not possible to infer the length of a repeat expression. Doing so would require moving the expression into a separate function generic over the array length.

```rust
fn foo() {
    // This errors due to `_` as a repeat count being unsupported
    let a: [_; 1] = [String::new(); _];
}
```

With `feature(generic_arg_infer)` all the previous examples compile. This should hopefully feel like something that should "obviously" be supported by Rust.
````